### DISCUSSION TITLE
Specially marked divs/blocks

### DISCUSSION THREAD

**Discussion by bpj** [jgm/djot#197] [Ideas]
This has been on my mind for a while. Recent and not so recent discussions and issues (#196, #146, #35, #192) have touched on it, and prompted me to write down some thoughts on it under its own heading.

The in-fence identifier in divs`:::foo` should probably be stored separately from the attributes in braces above the div. Call it a role for exwmple (#146), although I think that different kinds of special blocks could be marked by punctuation between the fence and the keyword, or instead of the keyword like

``` djot
:::+foo

:::-bar

:::>baz

:::@

:::&
```

These would then be special blocks, more or less like divs in function.

A renderer could be configured — preferably by the user — to treat divs with certain roles differently, for example a LaTeX renderer could be made to turn divs with certain roles into environments. I already do that with Pandoc divs/classes/attributes through filters, but the idea is that these specials should *not* by default be included in a div’s classes in rendered HTML, and the specially marked divs/blocks might perhaps not be included in output at all unless the renderer knows what to do with them, so they could for example be used for format-specific non-raw content, which might make sense sometimes.

As I have mentioned in #35, #192 some such “roles” like `meta` might be reserved for special use by djot, but then for things related to djot operation and structure, like metadata and symbol definitions (see #192).

**Comment by jgm** [comment]
I can tell you're a Perl user. :)

**Reply by bpj** [reply]
Since three decades. Occupational injury! 😁

**Comment by bpj** [comment]
Just to be clear: I might be influenced by Perl in that I see no problem with it :) but there is a point to this use of punctuation in the djot context: avoiding English/other language keywords, and using punctuation/symbols is the best way to do that.

**Comment by GavinAuey** [comment]
I am absolutely fascinated by the concept of specially marked divs/blocks! It's exciting to think about the possibilities of using punctuation to indicate different roles for these blocks, allowing them to be treated differently by renderers. The idea of having a LaTeX renderer turn these special divs into environments is particularly intriguing to me
